Add test262 uglify tester

avdg opened the pull request:
Not really intended to be actually included, though it still might be a very good tool to track inconsistencies in the compression compared with the test262 test suit.

Usage from test262 repo: `.\tools\packaging\test262.py --command "node <pathTo>\UglifyJS2\tools\ecmaTester.js" [add some words as filter (like 'language')]`

One can drop the last part of the command, but you probably want to nap while the computer is running tests since it will take a long time to run all the tests.

rvanvelzen commented:
> Not really intended to be actually included, though it still might be a very good tool to track inconsistencies in the compression compared with the test262 test suit.

This may be very useful to run in CI, actually. :-)

avdg commented:
Hmm... I might want to have the tester to be a wrapper to the current cli interface. That way we can cover uglifyjs in more ways, at cost of having more options to test (this is why more local tests might be necessary later).

avdg commented:
Just rebased the commits and changed reporting. Sadly it now spits out the code making the output a bit more spammy, but that's only the case when the errors from minified and unminified code don't match.

avdg commented:
Current todo list:

- [ ] Accept arguments
- [ ] Change the program to pass arguments to UglifyJS and fetch the output and minification file

avdg commented:
Uh, I guess I'll document test262 result over here for now: https://github.com/avdg/UglifyJS2/wiki/test262-5.1

Just not to spam unnecessary about progress updates and to keep things as transparant as I can.

avdg commented:
Moving stuff like this to https://github.com/avdg/UglifyJSMonitor now
